### DISCUSSION TITLE
CREATE or UPDATE an item through API

### DISCUSSION THREAD

**Discussion by TarekTolba1** [directus/directus#16296] [Archive]
I know that there is an API endpoint to create new items and another one to update an existing item. But is there a way to make a createOrUpdate request?

**Comment by 5m1Ly** [comment]
I'm assuming that you are creating your own unique identifier or allready have the identifier that is used within the row that you would like to update.

So if this is the case you could setup a Flow. For this go to the admin panel on port 8080, then navigate to the settings. Once your inside the settings go to Flows, now click the button with the + icon and make a new flow. At this point you'll see an setup panel to left enter a name for the flow here then click on the arrow.

When you are at the second step within the setup process you need to the following things,
1. select "Event Hook".
2. select "Filter (Blocking)" as the type.
3. select "item.create" as the scope. 
4. select the collection for which youre creating this flow.
5. now select "Data of Last Operation" as response body.

Once youre done with the setup click the button with the check icon. Now you should see a box with the "trigger" title, youre able to append operations to this box. Combine the condition operation with create, read and update operations to perform checks on the data which will then either create or update the collection.

If you want to know more about the condition operation, take a look at the [filter rules docs](https://docs.directus.io/reference/filter-rules.html) this might help with setting it up.

**Reply by TarekTolba1** [reply]
Thanks a lot :)

**Reply by valentinkorzun** [reply]
@TarekTolba1 Do you have an example how to do it?

**Reply by valentinkorzun** [reply]
@5m1Ly I don't understand a little. Does this trigger make sense when we subscribe to the event after the object has already been created in the database?

**Reply by 5m1Ly** [reply]
@valentinkorzun no it does but also doesn't make sense, the scenario described within the issue is what we call an "upsert query" in sql terminology. In short this means that you check if a specific piece of dat within a larger data set (usually the row id)  exists within the database. If the data doesn't exist you'll insert the data because it's new data. If the data exists you'll update the data because there are changes made within the data. But keep in mind that the needs to be new or has changed otherwise it wouldn't make sense to use an upsert query at all.

If you want to know more about upsert queries, i found this website may they explain it better 
https://www.cockroachlabs.com/blog/sql-upsert/

**Reply by jshay21** [reply]
hi @5m1Ly Do you have an example of this? **Combine the condition operation with create, read and update operations to perform checks on the data which will then either create or update the collection.**

**Reply by 5m1Ly** [reply]
Well @jshay21 actually not any more, because I don't have access to the system for which we where using this, but if you follow the step by step guide from above everything should work fine.

**Reply by 5m1Ly** [reply]
But you need some conditional operation, which will turn in into a bunch of spaghetti when it gets larger. So depending on the size of the use case you could also consider to make it simply by writing some code.

**Comment by rijkvanzanten** [comment]
See https://github.com/directus/directus/discussions/5706 🙂
